### PR TITLE
Create a simple JIT wrapper for regular Python functions

### PR DESCRIPTION
**Pull request by ehsantn**
## Changes included in this PR

Adds a `wrap_python` decorator that is both easier to use and also much faster than objmode (doesn't require compilation during runtime). This interface fits the common UDF case for calling regular Python especially well.

## Testing strategy

Added unit test and also tested the AI "the pile" data processing example (will be added in a separate PR).

## User facing changes

Adds `wrap_python` decorator.

## Checklist
- [ ] Pipelines passed before requesting review. To run CI you must include `[run CI]` in your commit message.
- [x] I am familiar with the [Contributing Guide](https://github.com/bodo-ai/Bodo/blob/main/CONTRIBUTING.md) 
- [x] I have installed + ran pre-commit hooks.

### CODE REVIEW

**Comment by njriasan** [comment]
@ehsantn Can you provide me some context on this change? Was this directly requested or has this been a known TODO for some time?

**Comment by ehsantn** [comment]
This came out of my work on AI examples and seeing the issues of objmode (interface, performance, reliability). This will have more feedback and iterations soon probably.

**Review by njriasan — APPROVED** [review]
LGTM! I don't have enough context on the reasoning for this change, but the implementation of this change seems reasonable.

The biggest possible concern is that Numba is looking to decrease the number of JIT decorators they maintain to reduce difficulties with their API. If there is a bug fix to say objmode that improves the quality of life I want to ensure that it seems likely this new API will pick it up.

**Comment by ehsantn** [comment]
I haven't seen activity on the objmode side but will monitor.

File: pixi.toml [hunk 1]
```diff
@@ -104,6 +104,8 @@ snowflake-connector-python = "*"
 xlrd = "*"
 xlsxwriter = "*"
 openpyxl = "*"
+datasets = "3.2.0.*"
+transformers = "4.47.1.*"
```
Comment (scott-routledge2) [inline pixi.toml:108]: I'm assuming these were for the AI examples? Are they needed at this point?

**Review by scott-routledge2 — APPROVED** [review]
LGTM, thanks @ehsantn !

File: pixi.toml (hunk 1 above)
Comment (ehsantn) [reply pixi.toml:108]: Got added by mistake. Will remove now.

Comment (ehsantn) [reply pixi.toml]: Done.

**Pull request merged**